salt/modules/file.py

```diff
@@ -1587,7 +1588,16 @@ def _get_line_indent(src, line, indent):
             break
         idt.append(c)
 
-    return ''.join(idt) + line.strip()
+    return ''.join(idt) + line.lstrip()
+
+
+def _set_line_eol(src, line):
+    '''
+    Add line ending
+    '''
+    regex = re.compile('((?<!\r)\n|\r(?!\n)|\r\n)$')
+    line_ending = regex.search(src).group()
```

Comment:
What happens on the final line of the file? Not all files end in a newline, and this regex wouldn't match a line without one:

```python
>>> regex = re.compile('((?<!\r)\n|\r(?!\n)|\r\n)$')
>>> regex.search('foo').group()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'group'
```

Reply:
I'll add guard statement.